graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/predicate/FieldEvaluator.java

```diff
@@ -146,21 +142,12 @@ public static FieldEvaluator forConnection(DataFetchingEnvironment environment)
                     .build();
 
             // Extract selection set on "{ nodes }" or "{ edges { node } }"
-            Map<String, MergedField> fields = null;
             if (environment.getSelectionSet() != null) {
-                MergedSelectionSet merged = environment.getSelectionSet().get();
-                fields = merged.getSubFields();
-                if (fields.containsKey("nodes")) {
-                    // First look in { nodes } selection set
-                    MergedField nodeFields = fields.get("nodes");
-                    return getField(fieldCollector.collectFields(parameters, nodeFields), name);
-                } else if (fields.containsKey("edges")) {
-                    // If no "nodes" was found, try to look into { edges { node } } selection set
-                    MergedField edgeFields = fields.get("edges");
-                    fields = fieldCollector.collectFields(parameters, edgeFields).getSubFields();
-                    if (fields.containsKey("node")) {
-                        return getField(fieldCollector.collectFields(parameters, fields.get("node")), name);
-                    }
+                DataFetchingFieldSelectionSet selectionSet = environment.getSelectionSet();
+                if (selectionSet.contains("nodes")) {
+                    return getField(selectionSet.getFieldsGroupedByResultKey("nodes/**"), name);
```

Comment:
It seems that retrieving fields from a selectionSet now supports glob pattern and supports recursing down fields and subfields, so not sure if fieldCollector is still needed here. These are all still untested (will try them out tomorrow) so not sure how valid of a solution this is yet.